controllers/postgresql/postgres/aws.go

```diff
@@ -111,5 +111,55 @@ func (c *awspg) DropRole(role, newOwner, database string) error {
 		}
 	}()
 
-	return c.pg.DropRole(role, newOwner, database)
+	return c.pg.DropRoleAndDropAndChangeOwnedBy(role, newOwner, database)
+}
+
+func (c *awspg) ChangeAndDropOwnedBy(role, newOwner, database string) error {
+	// On AWS RDS the postgres user isn't really superuser so he doesn't have permissions
+	// to REASSIGN OWNED BY unless he belongs to both roles
+	err := c.GrantRole(role, c.user)
+	// Check error
+	if err != nil {
+		// Try to cast error
+		pqErr, ok := err.(*pq.Error)
+		if !ok {
+			return err
+		}
+		if pqErr.Code == RoleNotFoundErrorCode {
+			return nil
+		}
+		if pqErr.Code != InvalidGrantOperationErrorCode {
+			return err
+		}
+	}
+
+	err = c.GrantRole(newOwner, c.user)
+	// Check error
+	if err != nil {
+		// Try to cast error
+		pqErr, ok := err.(*pq.Error)
+		if !ok {
+			return err
+		}
+
+		if pqErr.Code == RoleNotFoundErrorCode {
+			// The group role does not exist, no point of granting roles
+			c.log.Info(fmt.Sprintf("not granting %s to %s as %s does not exist", role, newOwner, newOwner))
```

Comment:
Maybe change log level to Warn ? Or RoleNotFound is a regular case ?

Reply:
Yes maybe but as this code have been copy pasted upper too, I think this must be changed in another PR. What do you think ?

Reply:
agreed :+1: